lib/autoload.fish

```diff
@@ -1,15 +1,22 @@
 # SYNOPSIS
-#   autoload <path [path...]>
+#   autoload <path>...
 #
 # OVERVIEW
 #   Autoload a function or completion path. Add the specified list of
 #   directories to $fish_function_path. Any `completions` directories
 #   are correctly added to the $fish_complete_path.
+#
+#   Returns 0 if one of the paths exist.
+#   Returns 1 if all paths are missing.
 
 function autoload -d "autoload a function or completion path"
+  set -l path_exist "false"
+
   for path in $argv
+    set -l dest fish_function_path
+
     if test -d "$path"
-      set -l dest fish_function_path
+      set path_exist "true"
```

Comment:
You can just `set -l path_exist` and in the end `set -q path_exist; and return 0; or return 1`.